### PR TITLE
OHRM5X-823: Update Password Screen Added

### PR DESCRIPTION
**Pull request by VageeshOrangeHRM**
This PR Contains following changes:
1. Password update End point added.
2. Integration test added for Password update API.
3. Data group permission and role permission added to Password  update API.
4. OHRM5X-792: Password update screen added.

### CODE REVIEW

File: symfony/client/src/orangehrmCorePlugin/pages/updatePassword/UpdatePassword.vue [hunk 1]
```diff
@@ -0,0 +1,148 @@
+<!--
```
Comment (samanthajayasinghe) [inline symfony/client/src/orangehrmCorePlugin/pages/updatePassword/UpdatePassword.vue:1]: @VageeshOrangeHRM can you move all pages to `OrangehrmPimPlugin`

Comment (VageeshOrangeHRM) [reply symfony/client/src/orangehrmCorePlugin/pages/updatePassword/UpdatePassword.vue]: done

File: symfony/plugins/orangehrmPimPlugin/Api/UpdatePasswordAPI.php [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+    public const PARAMETER_CURRENT_PASSWORD = 'currentPassword';
+    public const PARAMETER_NEW_PASSWORD = 'newPassword';
+
+    public const PARAM_RULE_STRING_MAX_LENGTH = 64;
+
+    /**
+     * @return UserService|null
+     */
+    public function getSystemUserService(): ?UserService
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/Api/UpdatePasswordAPI.php:51]:
```php
    /**
     * @return UserService
     */
    public function getSystemUserService(): UserService
```

File: symfony/plugins/orangehrmPimPlugin/Api/UpdatePasswordAPI.php [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                        }
+                    ])
+                )
+            ),
+            $this->getValidationDecorator()->requiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_NEW_PASSWORD,
+                    new Rule(Rules::STRING_TYPE),
+                    new Rule(Rules::LENGTH, [null, self::PARAM_RULE_STRING_MAX_LENGTH]),
+                    new Rule(Rules::CALLBACK, [
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/Api/UpdatePasswordAPI.php:118]: will move this to Validation rules, so we can reuse it.

**Review by RajithaKumara — APPROVED** [review]

File: symfony/plugins/orangehrmPimPlugin/Api/UpdatePasswordAPI.php [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+                        function () {
+                            $newPassword = $this->getRequestParams()->getString(
+                                RequestParams::PARAM_TYPE_BODY,
+                                self::PARAMETER_NEW_PASSWORD
+                            );
+                            $uppercase = preg_match('@[A-Z]@', $newPassword);
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmPimPlugin/Api/UpdatePasswordAPI.php:124]: we may need to support unicode chars

**Pull request merged**